chiselFrontend/src/main/scala/chisel3/internal/firrtl/IR.scala

```diff
@@ -136,6 +146,43 @@ sealed case class KnownWidth(value: Int) extends Width {
   override def toString: String = s"<${value.toString}>"
 }
 
+object BinaryPoint {
+  def apply(x: Int): BinaryPoint = KnownBinaryPoint(x)
+  def apply(): BinaryPoint = UnknownBinaryPoint()
+}
+
+sealed abstract class BinaryPoint {
+  type W = Int
+  def max(that: BinaryPoint): BinaryPoint = this.op(that, _ max _)
+  def + (that: BinaryPoint): BinaryPoint = this.op(that, _ + _)
+  def + (that: Int): BinaryPoint = this.op(this, (a, b) => a + that)
+  def shiftRight(that: Int): BinaryPoint = this.op(this, (a, b) => 0 max (a - that))
+  def dynamicShiftLeft(that: BinaryPoint): BinaryPoint =
+    this.op(that, (a, b) => a + (1 << b) - 1)
+
+  def known: Boolean
+  def get: W
+  protected def op(that: BinaryPoint, f: (W, W) => W): BinaryPoint
+}
+
+sealed case class UnknownBinaryPoint() extends BinaryPoint {
```

Comment:
Hard to argue with you, but all the other classes in IR carry these methods.  I'd prefer to leave them in for symmetry.  Although I don't see any usages of these particular methods, they are in moderate use for other classes that have them